src/types/any.rs

```diff
@@ -951,7 +951,7 @@ macro_rules! implement_binop {
 
 impl<'py> PyAnyMethods<'py> for Bound<'py, PyAny> {
     #[inline]
-    fn is<T: AsPyPointer>(&self, other: &T) -> bool {
+    fn is<T>(&self, other: &Bound<'py, T>) -> bool {
```

Comment:
I wonder, should we make the argument `T: AsRef<PyObject>`? That would accept a wider variety of types than just `Bound<'py, T>` and could be the same for `Py<T>` too.

The downside is we might need to add some `AsRef` implementations first 😂

Reply:
Great idea! I think this makes a lot of sense. I've added impls for `Bound`, `Py` and `Borrowed` (i guess these are the most common ones). Noteworthy, this breaks the inference of the existing `AsRef<Bound<PyAny>> for Bound<T>` but thats kind of a misuse anyway IMO, `as_any` is much clearer in non-generic context, so I would be fine with it.

Additionally we could add impls for `PyRef` and `PyRefMut`, but they will have the same inference breakage reagarding `are_ref <-> as_super`

Reply:
I think this looks good to me 👍 

I am ok with deferring `PyRef` and `PyRefMut` impls for another time. I think we've got enough possible breakage / changes coming to those types anyway and we can do this then.